### PR TITLE
docs: complex table filters example

### PR DESCRIPTION
**Pull request by tibuurcio**
## Summary
Documentation work and a change to an ESLint rule.

This PR brings the Observability complex filters implementation in a simplified version as to showcase an example of complex table filters to be used in the library.

New structure:

image

## Testing Plan
- [X] Was this tested locally? If not, explain why.

## Reference Issue (For mParticle employees only. Ignore if you are an outside contributor)
- Closes https://go.mparticle.com/work/UNI-1145

### CODE REVIEW

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const preview: Preview = {
           ],
           'Candidate Components',
           'UX Patterns',
-          ['Table', ['Table', 'Filters']],
```
Comment (tibuurcio) [inline .storybook/preview.tsx:41]: This sorting is not working when taking stories into consideration. Need to dig deeper to understand why.

File: docs/UX Patterns/Table/Filters.stories.tsx [hunk 2]
```diff
@@ -1,189 +1,33 @@
...
+  Select,
+  Divider,
+  type ICollapseProps,
+  ConfigProvider,
+  Collapse,
+  Checkbox,
 } from 'src/components'
 import { DatePickerWithDisabledYears } from 'src/components/data-entry/DatePicker/DatePicker.stories'
 import { SelectWithRangePicker } from 'docs/Candidate Components/Directory/Date Range Filter/SelectWithRangePicker'
-
```
Comment (tibuurcio) [inline docs/UX Patterns/Table/Filters.stories.tsx:22]: Extracted everything table related that would be re-used into `TableStoryUtils.tsx`

File: .storybook/preview.tsx [hunk 3]
```diff
@@ -1,9 +1,16 @@
 import type { Preview } from '@storybook/react'
+// import type { IndexEntry } from '@storybook/types'
 
 const preview: Preview = {
   parameters: {
     layout: 'centered',
     options: {
+      // TODO https://mparticle-eng.atlassian.net/browse/UNI-1214
```
Comment (tibuurcio) [inline .storybook/preview.tsx:8]: Tried implementing this on this PR but it's more complex than I was expecting, to pulled it out.

File: docs/UX Patterns/Table/TableStoryUtils.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
+import type { ReactNode } from 'react'
```
Comment (tibuurcio) [inline docs/UX Patterns/Table/TableStoryUtils.tsx:1]: Essencially pulled from Observability example, but kept everything in a single file and simplified a lot of it. No logic included, only UI.

File: src/components/data-display/Collapse/Collapse.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { Collapse as AntCollapse } from 'antd'
-import { type CollapseProps as AntCollapseProps } from 'antd'
+import type { CollapseProps as AntCollapseProps } from 'antd'
```
Comment (tibuurcio) [inline src/components/data-display/Collapse/Collapse.tsx:2]: suggested by eslint

**Review by gabyzif — APPROVED** [review]

**Pull request merged**